crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -926,15 +953,15 @@ where
                         %piece_offset,
                         "Failed to write piece into cache"
                     );
+                    caches.stored_pieces.insert(old_key, offset);
                 } else {
                     trace!(
                         %cache_index,
-                        %old_piece_index,
```

Comment:
Why was this line removed? It was important for debugging purposes to track replacement of pieces.

Reply:
there is not save the stored piece_index for now, Let me add a todo and think about what the best way to do this is

Reply:
Not sure what you mean, but we do evict one of the previously stored pieces in this branch and we need to log which one.